runtime/resolvers/metrics_sql.go

```diff
@@ -0,0 +1,245 @@
+// The compiler preserves templating in the SQL, allowing the regular SQL resolver to handle SQL templating rules.
+func newMetricsSQL(ctx context.Context, opts *runtime.ResolverOptions) (runtime.Resolver, error) {
+	props := &metricsSQLProps{}
+	if err := mapstructure.Decode(opts.Properties, props); err != nil {
+		return nil, err
+	}
+
+	if props.SQL == "" {
+		return nil, errors.New(`metrics SQL: missing required property "sql"`)
+	}
+
+	ctrl, err := opts.Runtime.Controller(ctx, opts.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	olap, release, err := opts.Runtime.OLAP(ctx, opts.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	compiler := &metricsSQLCompiler{
+		instanceID:     opts.InstanceID,
+		ctrl:           ctrl,
+		sql:            props.SQL,
+		dialect:        olap.Dialect(),
+		args:           opts.Args,
+		userAttributes: opts.UserAttributes,
+	}
+
+	sql, connector, refs, err := compiler.compile(ctx)
+	if err != nil {
+		release()
+		return nil, err
+	}
+
+	// Build the options for the regular SQL resolver
+	sqlResolverOpts := &runtime.ResolverOptions{
+		Runtime:    opts.Runtime,
+		InstanceID: opts.InstanceID,
+		Properties: map[string]any{
+			"connector": connector,
+			"sql":       sql,
+		},
+		Args:           opts.Args,
+		UserAttributes: opts.UserAttributes,
+		ForExport:      opts.ForExport,
+	}
+
+	return newSQLWithRefs(ctx, sqlResolverOpts, refs)
+}
+
+var (
+	// sqlIdentifier is regex pattern to identify a SQL identifier. The identifier may be wrapped in double quotes.
+	// Additionally if double quotes are present in identifier, it is escaped with additional double quotes.
+	sqlIdentifier = `[a-zA-z_][a-zA-Z0-9_]*|"(?:[^"]|"")*"`
+
+	// aggregateRegex is regex pattern to identify an AGGREGATE function in SQL.
+	aggregateRegex = regexp.MustCompile(fmt.Sprintf(`(?i)AGGREGATE\((?:(%s)\.)?(%s)\)`, sqlIdentifier, sqlIdentifier))
+
+	// fromRegex is regex pattern to identify a FROM clause in SQL.
+	fromRegex = regexp.MustCompile(fmt.Sprintf(`(?i)FROM\s+(%s)`, sqlIdentifier))
+)
+
+// metricsSQLCompiler parses a metrics SQL query and compiles it to a regular SQL query.
+// Metrics SQL is a superset of SQL that supports querying Rill's metrics views.
+// The syntax is inspired by Calcite's measure columns: https://issues.apache.org/jira/browse/CALCITE-4496.
+//
+// This is a simple implementation that uses regular expressions. It does not support all SQL features.
+// It works by:
+//
+// 1. Expanding AGGREGATE(measure) into actual aggregate expressions from the metrics view definition.
+// 2. Converting "FROM metrics_view" clauses to nested SELECTs on the underlying table with filters based on the metrics view's security policy.
+//
+// Example transformation:
+// - Input: SELECT dim1, AGGREGATE(measure1) FROM metrics_view GROUP BY dim1
+// - Output: SELECT col1 AS dim1, SUM(col2) AS measure1 FROM underlying_model GROUP BY dim1
```

Comment:
The output is not how it is expanded right now. 
1. `dimension` is not yet supported. 
2. `AGGREGATE(measure1)` is expanded as just `SUM(col2)` right now.
Though both can be fixed with the planned changes for support for dimension expressions.

Reply:
Changed to a `TODO` :)